src/ripple/app/tx/impl/InvariantCheck.cpp

```diff
@@ -0,0 +1,130 @@
+//------------------------------------------------------------------------------
+/*
+  This file is part of rippled: https://github.com/ripple/rippled
+  Copyright (c) 2012-2016 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/tx/impl/InvariantCheck.h>
+#include <ripple/basics/Log.h>
+
+namespace ripple {
+
+void
+XRPNotCreated::visitEntry(
+    uint256 const& /*index*/,
+    bool isDelete,
+    std::shared_ptr <SLE const> const& before,
+    std::shared_ptr <SLE const> const& after)
+{
+    if(before)
+    {
+        switch (before->getType())
+        {
+        case ltACCOUNT_ROOT:
+            drops_ -= (*before)[sfBalance].xrp().drops();
+            break;
+        case ltPAYCHAN:
+            drops_ -= ((*before)[sfAmount] - (*before)[sfBalance]).xrp().drops();
+            break;
+        case ltESCROW:
+            drops_ -= (*before)[sfAmount].xrp().drops();
+            break;
+        default:
+            break;
```

Comment:
If we want to add this check for known types of LE, I'd be inclined to create a separate check for this rather than wrapping it into this one. @JoelKatz do you think it's worth adding a LedgerEntryType-is-valid invariant check?